File: py/examples/basic.py
from imbrace import ImbraceClient, extract_api_key
from dotenv import load_dotenv
import os

load_dotenv()

# Initialize client with explicit API Key (e.g. from environment)
api_key = os.environ.get("IMBRACE_API_KEY")
if not api_key:
    print("Warning: IMBRACE_API_KEY not set")

client = ImbraceClient(
    api_key=api_key,
)


# import httpx
# res = httpx.get(
#     "https://app-gatewayv2.imbrace.co/auth/key",
#     headers={"X-Api-Key": "your-raw-key"}
# ).json()
# client = ImbraceClient(api_key=extract_api_key(res))

# List sessions
sessions = client.sessions.list()
print("Sessions:", sessions)

# Create session and send a message
session = client.sessions.create(directory=os.getcwd())
response = client.messages.send(
    session["id"],
    parts=[{"type": "text", "text": "Hello from Imbrace Python SDK!"}],
    directory=os.getcwd(),
)
print("Response:", response)

client.close()

File: ts/examples/basic.ts
// Imbrace TS SDK — Basic usage with real Imbrace Gateway
import { ImbraceClient, extractApiKey } from "../src/index.js"

// Initialize client with explicit API Key
const client = new ImbraceClient({
  apiKey: process.env.IMBRACE_API_KEY, // Explicitly pass from env if desired
})

// Or --- Option 2: From the auth endpoint response ---
// const authResponse = await fetch("https://app-gatewayv2.imbrace.co/auth/key")
//   .then(r => r.json())
// const client = new ImbraceClient({
//   apiKey: extractApiKey(authResponse), // gets authResponse.apiKey.apiKey
// })

// Optional: verify server is reachable first
await client.init()

// List all sessions
const sessions = await client.sessions.list()
console.log("Sessions:", sessions)

// Create a session and send a prompt
const session = await client.sessions.create({ directory: process.cwd() })
const response = await client.messages.send(session.id, {
  parts: [{ type: "text", text: "Hello, Imbrace!" }],
  directory: process.cwd(),
})

console.log("Response:", response)
